sound/soc/sof/topology.c

```diff
@@ -473,142 +473,155 @@ static enum sof_comp_type find_process_comp_type(enum sof_ipc_process_type type)
 struct sof_topology_token {
 	u32 token;
 	u32 type;
-	int (*get_token)(void *elem, void *object, u32 offset);
+	void (*get_token)(void *elem, void *object, u32 offset);
 	u32 offset;
+	u32 (*find_token)(void *elem);
```

Comment:
you are so very right. I should simply change what we pass in parse_string_tokens() to elem->string instead of elem and just reuse get_token. Thanks for suggesting this.